spec/cc/workspace/path_tree_benchmark_spec.rb

```diff
@@ -0,0 +1,90 @@
+require "spec_helper"
+
+class CC::Workspace
+  describe PathTree do
+    include FileSystemHelpers
+
+    MAX_FILTER_SECONDS = 10
+    NUM_FILES = 10_000
+    AVG_FILES_PER_DIR = 20
+    REPO_DEPTH = 30
+    NUM_EXCLUDES = 50
+
+    it "performs acceptably" do
+      within_temp_dir do
+        make_files
+        exclude_paths = make_exclude_paths
+
+        start_time = Time.now
+        tree = PathTree.new(".")
+        tree.exclude_paths(exclude_paths)
+        elapsed = Time.now - start_time
```

Comment:
Yes, probably -- I also turned up some unexpected things about the size of the generated workspace while porting this to master (master did blow well past the 10 second mark at about 2 minutes, FWIW), so I think this spec needs some attention overall.